src/protobuf-net/Meta/MetaType.cs

```diff
@@ -2482,5 +2482,13 @@ static string CommentSuffix(ProtoReservedAttribute reservation)
                 return " (" + comment.Trim() + ")";
             }
         }
+
+        internal MetaType GenerateGenericType(Type type)
+        {
+            var metaType = new MetaType(model, type, factory);
+            metaType.SetSurrogate(surrogateType.GetGenericTypeDefinition());
+            metaType._fields = _fields;
+            return metaType;

```

Comment:
Pretty sure this part will need further refinement. I'm assigning `_fields` versus duplicating the field data as it should be an exact match of data - I can't think of a way this wouldn't work. Happy to instead iterate over the data and manually call `AddField` etc.

Also there may be other important properties to set from one `MetaType` instance to the other here.

Reply:
I updated the code to include additional properties on `MetaType` that seemed relevant but I'd need your expertise to confirm my approach and whether there are any other considerations I should be aware of.